stdlib/public/Concurrency/DispatchExecutor.swift

```diff
@@ -161,20 +161,29 @@ enum DispatchClockID: CInt {
 @available(SwiftStdlib 6.2, *)
 extension DispatchExecutorProtocol {
 
+  func clamp(_ components: (seconds: Int64, attoseconds: Int64))
+    -> (seconds: Int64, attoseconds: Int64) {
+    if components.seconds < 0
```

Comment:
What happens if `components.seconds == 1` but `components.attoseconds` is so negative that the compound is still less than 0?

`(seconds: 1, attoseconds: -2_000_000_000_000_000_000)` or so?

Reply:
> (seconds: 1, attoseconds: -2_000_000_000_000_000_000)

or ofc `(seconds: 1, attoseconds: Int64.min)` which I believe is -9.something seconds

Reply:
```
@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
extension Duration {

    /// The composite components of the `Duration`.
    ///
    /// This is intended for facilitating conversions to existing time types. The
    /// attoseconds value will not exceed 1e18 or be lower than -1e18.
    @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
    public var components: (seconds: Int64, attoseconds: Int64) { get }
}
```

A user can just set those fields independently, right?

Reply:
No. If you try to use the `init()` that takes those components, you'll find it adds and multiplies in order to get a count of just attoseconds. And `components` is `get`-only.

Reply:
Ah I see, thank you!